File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        pascal = []
        # intialize the pascal array all with 1 value
        pascal = [ [1] * (i + 1) for i in range(numRows)]
        
        # Build the pascal triangle
        for i in range(2, numRows): # start from second row
            for j in range(1, i):   # start form second column upto second of last number
                pascal[i][j] = pascal[i - 1][j - 1] + pascal[i - 1][j]
        
        return pascal

File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector < vector<int> > pascal;
        for (int i = 0; i < numRows; i++)
        {
           vector <int> row(i + 1, 1);
            pascal.push_back(row);
        }

        for (int i = 2; i < numRows; i++)
        {
            for (int j = 1; j < i; j++)
            {
               pascal[i][j] = pascal[i - 1][j-1] + pascal[i - 1][j];
            }
        }
            return pascal;
    }
};